### PR TITLE
Cu 86bxu2a8m/be refactor the group contributions model with the new db design

### PR DESCRIPTION
**Pull request by Don-christo**
**Description of PR**

This pull request includes refactoring of the group models and endpoints. It fixes the issue with the `GroupWallet` model where the `ownerId` attribute was changed to `groupId`. It also updates the `fundGroupWallet` endpoint to use the `groupId` attribute instead of `ownerId`. Additionally, it removes the `GroupContributors` and `GroupTransactions` models as they are no longer needed.

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [x] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_
Screen Shot 2024-03-19 at 3 00 33 PM
---

**How to test**  
Use the request payload from the screenshot added and call this endpoint on postman: `api/v1/groups/createGroup`

### CODE REVIEW

File: apps/server/src/controllers/groupControllers/helpers.ts [hunk 1]
```diff
@@ -1,16 +1,170 @@
...
+//   return {
+//     id: v4(),
+//     userId: props.userId,
+//     groupId: props.groupId,
+//     groupTitle: props.groupTitle,
+//     amountContributed: 0,
+//     totalAmountWithdrawnByUser: 0,
+//     dateOfLastContribution: null,
+//   }
+// }
```
Comment (KristofaJosh) [inline apps/server/src/controllers/groupControllers/helpers.ts:22]: ??

File: apps/server/src/models/association_bak
Comment (KristofaJosh) [inline apps/server/src/models/association_bak:1]: please delete this file

File: apps/server/src/controllers/groupControllers/helpers.ts [hunk 2]
```diff
@@ -1,16 +1,170 @@
...
+          console.log(
+            'Cashout email has been sent to your email if you have an account with us: ' +
+              info.response
+          )
+        }
+      }
+    )
+  } else {
+    console.log('No next beneficiary to notify.')
+  }
```
Comment (KristofaJosh) [inline apps/server/src/controllers/groupControllers/helpers.ts:123]: I think we have an emailFactory function that handles this

Comment (KristofaJosh) [reply apps/server/src/controllers/groupControllers/helpers.ts:123]: https://github.com/Ajo-Savings-Project/ajo-savings/blob/main/apps/server/src/utils/createEmailFactory.ts

Comment (KristofaJosh) [reply apps/server/src/controllers/groupControllers/helpers.ts:123]: use as 

```
const {sendMail} = createEmail({
  to,
  subject,
  templatePath,
  templateContext,
})

await sendMail()
```

**Review by KristofaJosh — CHANGES_REQUESTED** [review]
LGTM!

make those changes and re-request a review

File: apps/server/src/controllers/groupControllers/helpers.ts (hunk 1 above)
Comment (Don-christo) [reply apps/server/src/controllers/groupControllers/helpers.ts:22]: Mistake boss

File: apps/server/src/models/association_bak
Comment (Don-christo) [reply apps/server/src/models/association_bak:1]: Okay boss

File: apps/server/src/controllers/groupControllers/helpers.ts (hunk 2 above)
Comment (Don-christo) [reply apps/server/src/controllers/groupControllers/helpers.ts:123]: Okay boss

File: apps/server/src/controllers/groupControllers/helpers.ts [hunk 3]
```diff
@@ -1,16 +1,132 @@
...
+    isAdmin: boolean
+  }>
+}
+
+export const createNewMember = async (props: Props) => {
+  const user = await getUserWithId(props.userId)
+
+  if (user) {
+    let sequence
+    const MAX_SEQUENCE_VALUE = 1000000
```
Comment (KristofaJosh) [inline apps/server/src/controllers/groupControllers/helpers.ts:26]: This can be outside the function scope

**Review by KristofaJosh — DISMISSED** [review]

File: apps/server/src/models/groups.ts [hunk 4]
```diff
@@ -104,13 +99,16 @@ Groups.init(
...
+  foreignKey: 'adminId',
+  as: 'admin',
+  targetKey: 'id',
 })
 
-Groups.belongsTo(GroupWallet, {
-  foreignKey: 'id',
-  as: 'groupWallet',
-  targetKey: 'ownerId',
+Users.hasMany(Groups)
```
Comment (Jidekas) [inline apps/server/src/models/groups.ts:107]: specify the foreignKey in this association as well in order to eliminate the strange insertions in the DB

File: apps/server/src/models/groups.ts [hunk 5]
```diff
@@ -104,13 +99,16 @@ Groups.init(
...
+
+GroupWallet.belongsTo(Groups)
```
Comment (Jidekas) [inline apps/server/src/models/groups.ts:109]: specify the foreignKey in this association as well

**Review by Jidekas — CHANGES_REQUESTED** [review]

File: apps/server/src/models/groups.ts (hunk 4 above)
Comment (Don-christo) [reply apps/server/src/models/groups.ts]: Done. Thank you.

File: apps/server/src/models/groups.ts (hunk 5 above)
Comment (Don-christo) [reply apps/server/src/models/groups.ts]: Done. Thank you.

File: apps/server/src/controllers/groupControllers/helpers.ts (hunk 3 above)
Comment (Don-christo) [reply apps/server/src/controllers/groupControllers/helpers.ts]: Okay, boss. Done. Thank you.

**Review by Jidekas — APPROVED** [review]

**Review by KristofaJosh — APPROVED** [review]

**Pull request merged**